### DISCUSSION TITLE
##[error]Unable to locate executable file: 'tar'.

### DISCUSSION THREAD

**Discussion by KristofKuli** [allure-framework/allure2#2446] [Bug Reports]
Since Allure extension has been updated to version 1.2.0, the target path configuration has been changed from "reportDir" to "testResultsDir" and we are getting the following error message:
`##[error]Unable to locate executable file: 'tar'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.`

We are not sure, if the issue is with the target path definition or with the new version itself. The full log can be found below.

```
Starting: Publish allure report
==============================================================================
Task         : Publish Allure Report
Description  : Task for generate and publish Allure Report
Version      : 1.2.0
Author       : Qameta Software Inc.
Help         : Generate and publish Allure Report
==============================================================================
AllureReport input: {
  allureVersion: '2.26.0',
  testResultsDir: 'D:\\build\\V1\\_w\\1\\s\\res-cukes-rest-tests-arena-testsandbox\\target\\site\\allure-maven-plugin',
  reportDir: 'D:\\build\\V1\\_w\\1\\s\\c9462aee-fb03-467c-83ca-7564fe9fbc9a'
}
Found 10 test results files.
Downloading: https://github.com/allure-framework/allure2/releases/download/2.26.0/allure-2.26.0.tgz
Extracting archive
Error: Unable to locate executable file: 'tar'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.
    at Object._which (D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\node_modules\azure-pipelines-task-lib\internal.js:323:23)
    at new ToolRunner (D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\node_modules\azure-pipelines-task-lib\toolrunner.js:31:29)
##[error]Unable to locate executable file: 'tar'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also verify the file has a valid extension for an executable file.
    at Object.tool (D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\node_modules\azure-pipelines-task-lib\task.js:1282:14)
    at D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\tool.js:468:21
    at Generator.next (<anonymous>)
    at D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\tool.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\tool.js:4:12)
    at extractTar (D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\node_modules\azure-pipelines-tool-lib\tool.js:463:12)
    at D:\build\V1\_w\_tasks\PublishAllureReport_867c4cd2-7a10-11ed-a1eb-0242ac120189\1.2.0\index.js:50:53
Finishing: Publish allure report
```

**Comment by epszaw** [comment]
There wasn't any change related to the code that downloads allure and creates an executable from it.

The process doesn't have access to `tar` executable, so it could happen when the PATH variable was modified. Did you change them in the pipeline configuration?

**Comment by KristofKuli** [comment]
Hi Konstatin,
There's no modification is being done in the pipeline, regarding the PATH variable.

**Comment by epszaw** [comment]
Replaced azure tool utility by the `tar` node module in the `1.2.2` version of the extension. Could you try to run the pipeline once more and verify the result?

**Comment by KristofKuli** [comment]
With the new version, there's no errors and the pipelines finished successfully. However, the Tests tab displays all the passed tests, while navigating to the Allure Report tab, it doesn't show any result.
Tests:
 
![image](https://github.com/allure-framework/allure2/assets/103577855/4a80f936-3585-4a61-b78d-b0baabdd6d91)

Allure Report:
![image](https://github.com/allure-framework/allure2/assets/103577855/6c8dd814-d1e7-46db-94ff-2795599e7687)

The generated reports are available as build artifacts.

**Reply by epszaw** [reply]
Could you check `testResultsDir` input value? Does it point to a directory with Allure Results?

**Comment by KristofKuli** [comment]
Sorry, I have missed to add this part:
What can be seen in the logs:

testResultsDir: 'D:\\build\\V1\\_w\\1\\s\\res-cukes-rest-tests-arena-testsandbox\\target\\site\\allure-maven-plugin',
reportDir: 'D:\\build\\V1\\_w\\1\\s\\<ID>'
Seems reportDir is not correct, because report is generated under
'D:\\build\\V1\\_w\\1\\s\\res-cukes-rest-tests-arena-testsandbox\\target\\site\\allure-maven-plugin'

**Reply by epszaw** [reply]
`reportDir` generates dynamically for each task run.

The extension generates the report automatically, so you need just to pass the correct path to allure results directory. 

In your case, just check, where the directory is located locally and then, pass the relative path from the project's root to the directory into `testResultsDir`.

**Comment by KristofKuli** [comment]
From logs:
Report successfully generated to D:\build\V1\_w\1\s\res-cukes-rest-tests-arena-testsandbox\target\site\allure-maven-plugin

Task:
    - task: PublishAllureReport@1
      displayName: 'Publish allure report'
      condition: succeededOrFailed()
      inputs:
       testResultsDir: 'res-cukes-rest-tests-arena-testsandbox/target/site/allure-maven-plugin'

**Reply by epszaw** [reply]
Where do you store your test results locally in the project?

**Reply by KristofKuli** [reply]
Hi Konstantin,

I have received the following answer:
"The real generated report is added as build artifact, and you can download it and see the real results:
folder "Allure_report_ALL"
![image](https://github.com/allure-framework/allure2/assets/103577855/29ded1f6-fe13-4e43-abe5-199d828a4f3c)

**Reply by epszaw** [reply]
Could you describe your pipeline workflow? The extension looks for `.json` results files, not already generated report.

**Reply by KristofKuli** [reply]
The project is using separated test modules, you can find and example below.
```
- job: CARE_TEST_1
    steps:
    - task: Maven@3
      displayName: 'CARE run E2E tests task'
    - task: PublishPipelineArtifact@1
      condition: succeededOrFailed()
      inputs:
        targetPath: res-cukes-rest-tests-care-testsandbox/target/allure-results
        artifactName: CARE_1_allure_results
```
This is description I have received:
"It is pipeline Jobs related to report. As you can see separate tests modules are running, all generated reports files are saved as artifacts, you can check them. Then all files are copied to one directory: 'res-cukes-rest-tests-arena-testsandbox/target/allure-results'  and report generation are run, all Jobs are successful"

**Reply by KristofKuli** [reply]
Hi Konstantin,

Is there any update on this?

**Reply by KristofKuli** [reply]
Hi Konstantin,

Is there any update on this?

**Reply by KristofKuli** [reply]
Hi @epszaw,

Is there any update on this?

**Reply by epszaw** [reply]
Sorry, I was working on another important thing these weeks.

You don't need to handle `allure-results` directory by another task. Just pass a correct `resultsDir` input into the `PublishAllureReport` task. When paths will match – Allure Report binary will generate html file and automatically upload it.

Try to add a custom task which will log a path to your `allure-results` into the pipeline's output. Then, compare your `resultsDir` input with the path you receive in the output, they should be same.

**Reply by trushechinam** [reply]
Please let me know on what task I am hangling aditionally allure-results directory?
From my pipeline:

  - task: Maven@3
      displayName: 'ARENA_1 Generate Allure report'
      condition: succeededOrFailed()
      inputs:
        mavenPomFile: 'res-cukes-rest-tests-arena-testsandbox/pomFirst.xml'
        javaHomeOption: 'JDKVersion'
        jdkVersionOption: '1.11'
        jdkArchitectureOption: 'x64'
        goals: 'allure:report'

Gives such result:
![2024-04-23_10-22-23](https://github.com/allure-framework/allure2/assets/31212831/64170be1-7ea5-424a-8b84-d84f53825f53)


To be sure that directory and generated report is correct I have created another task, to save report data:

    - task: PublishPipelineArtifact@1
      displayName: 'Publish all files'
      condition: succeededOrFailed()
      inputs:
        targetPath: res-cukes-rest-tests-arena-testsandbox/target/site/allure-maven-plugin
        artifactName: Allure_report_All

![2024-04-23_10-25-05](https://github.com/allure-framework/allure2/assets/31212831/bb191338-79ba-40a8-bdbf-c2f4ba8e98ba)


I am able to download and open report locally
And it is what I see

![2024-04-23_10-26-08](https://github.com/allure-framework/allure2/assets/31212831/11ce2ee9-7308-486e-a284-b8c46b5820ba)


It means that report was generated correctly and issue is with task to publish it, directly this one:

 - task: PublishAllureReport@1
      displayName: 'Publish allure report'
      condition: succeededOrFailed()
      inputs:
       testResultsDir: 'res-cukes-rest-tests-arena-testsandbox/target/site/allure-maven-plugin'

What for some reason do not show correct report data

**Reply by trushechinam** [reply]
Could be "testResultsDir" in my case should have another path, could be it should contain index.html file? Because it seems it doesn't understand the path specified, and do not find there results

**Reply by ciconq** [reply]
Hey @trushechinam, maybe try adding this task for fetching the actual `allure-results` path, I am using it before the publish Allure report task.

```
      - task: PowerShell@2
        displayName: 'Get allure-results path'
        condition: succeededOrFailed()
        inputs:
          targetType: 'inline'
          script: |
            $selectedFolder = Get-ChildItem -Path $(System.DefaultWorkingDirectory) -Recurse -Filter "allure-results" -Directory | Select-Object -First 1
            Write-Output "Selected folder: $($selectedFolder.FullName)"
            Write-Output "##vso[task.setvariable variable=allureResultsPath]$($selectedFolder.FullName)"

      - task: PublishAllureReport@1
        displayName: 'Publish Allure report'
        condition: succeededOrFailed()
        inputs:
          allureVersion: '2.27.0'
          testResultsDir: '$(allureResultsPath)'
```

**Reply by trushechinam** [reply]
Hello, I have added tasks what you have proposed
What I get:
Selected folder: D:\build\V1\_w\1\s\res-cukes-rest-tests-arena-testsandbox\target\allure-results

And finally report have been generated.
I would say that it is `PublishAllureReport@1` task issue that it doesn't understand `testResultsDir: 'res-cukes-rest-tests-arena-testsandbox/target/site/allure-maven-plugin' ` as
- it used to work before
- all other tasks are working correctly with such path specified.

Thank you for help,
Marina.

**Reply by KristofKuli** [reply]
Hi All,

Thank you for the help. Marina confirmed that the extension if working fine for her, so we can close this Issue.

Best regards,
Kristof
